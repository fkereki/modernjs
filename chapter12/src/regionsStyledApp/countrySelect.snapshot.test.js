/* @flow */

import React from "react";
import TestRenderer from "react-test-renderer";

import { CountrySelect } from "./countrySelect.component";

const fakeDeviceData = {
    isTablet: false,
    isPortrait: true,
    height: 1000,
    width: 720,
    scale: 1,
    fontScale: 1
};

describe("CountrySelect", () => {
    it("renders correctly when loading, with no countries", () => {
        const tree = TestRenderer.create(
            <CountrySelect
                deviceData={fakeDeviceData}
                loading={true}
                currentCountry={""}
                onSelect={() => null}
                getCountries={() => null}
                list={[]}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders correctly a countries dropdown", () => {
        const tree = TestRenderer.create(
            <CountrySelect
                deviceData={fakeDeviceData}
                currentCountry={""}
                loading={false}
                onSelect={() => null}
                getCountries={() => null}
                list={[
                    {
                        countryCode: "UY",
                        countryName: "Uruguay"
                    },
                    {
                        countryCode: "AR",
                        countryName: "Argentina"
                    },
                    {
                        countryCode: "BR",
                        countryName: "Brazil"
                    }
                ]}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
