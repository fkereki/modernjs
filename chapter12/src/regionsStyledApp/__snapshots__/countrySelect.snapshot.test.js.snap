// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelect renders correctly a countries dropdown 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Country:
  </Text>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Select a country:",
            "textColor": undefined,
            "value": "",
          },
          Object {
            "label": "Argentina",
            "textColor": undefined,
            "value": "AR",
          },
          Object {
            "label": "Brazil",
            "textColor": undefined,
            "value": "BR",
          },
          Object {
            "label": "Uruguay",
            "textColor": undefined,
            "value": "UY",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`CountrySelect renders correctly when loading, with no countries 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Loading countries...
  </Text>
</View>
`;
